piccolo_admin/endpoints.py

```diff
@@ -431,12 +433,17 @@ def __init__(
         allowed_hosts: t.Sequence[str] = [],
         debug: bool = False,
         sidebar_links: t.Dict[str, str] = {},
+        mfa_provider: t.Optional[MFAProvider] = None,
```

Comment:
Bit of a cross reference without running it, is this intended to be typed as `t.Optional[t.Sequence[MFAProvider]]`? Or is the intention; for this release at least; to just provide support for setting up only a single provider?

I'm guessing based of the implementation at a glance that if someone were to create the relevant row entries in a table then they would still be able to authenticate it seems

Reply:
I was going back and forth on it.

For an initial attempt, just supporting a single MFA provider is easier, but then it backs us into a corner if we want to later support multiple MFA providers.

I'll change it so it's a `t.Optional[t.Sequence[MFAProvider]]`. That means:

- Worst case scenario we'll just raise a `ValueError` for now if more than one is provider passed in, saying support for multiple MFA providers is coming soon.
- Or just support multiple MFA providers straight away - the amount of extra work is pretty minimal. We just need an extra UI view which lets the user select which MFA provider they're setting up.

Reply:
Yea either choice is fine, just figured I'd call it out in the case it hadn't been considered. Possibly the following flow may help alleviate some of the needing to implement it pain while remaining future focused?
```python
class AdminRouter:
    def __init__(
        self,
        mfa_provider: t.Optional[t.Sequence[MFAProvider]] = None,
    ):
        self.mfa_providers: t.Optional[t.Sequence[MFAProvider]] = mfa_provider

        if self.mfa_providers:
            if len(self.mfa_providers) == 1:
                # Only one provider, skip the 'MFA selection' screen
                mfa_provider = self.mfa_providers[0]
                private_app.mount(
                    path="/mfa-setup/",
                    app=RateLimitingMiddleware(
                        app=mfa_setup(
                            provider=mfa_provider,
                            auth_table=self.auth_table,
                        ),
                        provider=rate_limit_provider,
                    ),
                )

            else:
                # More then one provider present, present
                # a selection screen for which one to set up
                raise ValueError(
                    "Piccolo does not currently support multiple MFA providers. This will be coming soon."
                )
```

Reply:
Yes, exactly - that would work nicely.

Reply:
For now, I'll raise a `ValueError` if more than one MFA provider is specified. I think we can support multiple providers in the future.